png_cicp_chunk_proposal.md

```diff
@@ -13,31 +13,33 @@ An existing W3C group note, [BT2100-in-PNG]  specifies an approach which is limi
 * extensible signaling of color space based on H.273
 * does not require the presence of iCCP chunk and embedded ICC profiles
 
+## Non Requirements
+* compatibility with HDR colorspace definitions in any other raster image format (eg. JPEG, AVIF)
+
 ## Strawman approach
-[H.273](https://www.itu.int/rec/T-REC-H.273/en) specifies a controlled vocabulary for the parameterization of
-color space information.
+[H.273](https://www.itu.int/rec/T-REC-H.273/en) specifies a controlled vocabulary for the parameterization of color space information.
 
-Define a cICP chunk that contains the 7 bytes necessary to carry the
-H.273 color space parameters:
+Define a `cICP` chunk that contains the 7 bytes necessary to carry the H.273 color space parameters:
 
-* COLPRIMS, 2 bytes, One of the ColourPrimaries enumerated values specified in Rec. ITU-T H.273 | [ISO/IEC 23091-2]
-* TRANSFC, 2 bytes, One of the TransferCharacteristics enumerated values specified in Rec. ITU-T H.273 | [ISO/IEC 23091-2]
-* VIDFRNG, 1 byte, Value of the VideoFullRangeFlag specified in Rec. ITU-T H.273 | [ISO/IEC 23091-2]
+* **COLPRIMS**, 2 bytes, One of the ColourPrimaries enumerated values specified in Rec. ITU-T H.273 | [ISO/IEC 23091-2]
+* **TRANSFC**, 2 bytes, One of the TransferCharacteristics enumerated values specified in Rec. ITU-T H.273 | [ISO/IEC 23091-2]
```

Comment:
Although these two fields are two bytes in the ColourInformationBox (the 'colr' box) in ISO BMFFL

```
    unsigned int(16) colour_primaries;
    unsigned int(16) transfer_characteristics;
```

ITU-T H.273 says:

> An 8-bit field should be adequate for representation of the ColourPrimaries code point.

and

> An 8-bit field should be adequate for representation of the TransferCharacteristics code point

And indeed both HEVC (H.265) and AV1 represent these with 8-bit syntax elements. See page 415 in Rec. ITU-T H.265 v7 (11/2019) and page 32 in _AV1 Bitstream & Decoding Process Specification, Version 1.0.0 with Errata 1_. (They are also 8-bit syntax elements in H.274, used by VVC.)

So we may want to use only one byte each for COLPRIMS and TRANSFC.

Also, VIDFRNG can be represented with one bit.

Reply:
Per 2021-02-24: Use one byte for color primaries, transfer function and matrix coefficient fields, matching H.273.

Reply:
Leave a `NOTE` that JPEG standards are 2 bytes vs. 1 here...